src/TSTransformer/util/offset.ts

```diff
@@ -16,6 +16,20 @@ export function offset(expression: luau.Expression, value: number) {
 	if (value === 0) {
 		return expression;
 	}
+	// special case to handle adding and removing the offset
+	if (expression.kind === luau.SyntaxKind.BinaryExpression) {
+		if (expression.right.kind === luau.SyntaxKind.NumberLiteral) {
+			if (expression.operator === "+" || expression.operator === "-") {
+				const rightValue = getLiteralNumberValue(expression.right)!;
+				if (rightValue + value === 0) {
+					return expression.left;
+				} else {
+					expression.right.value = (rightValue + value).toString();
```

Comment:
Luau nodes are meant to be immutable - I'm surprised the types don't actually reflect that.
Since we're returning the expression on the next line anyways, it should be easy to just construct a new one.